src/main/java/org/jenkinsci/plugins/vsphere/builders/ReconfigureCpu.java

```diff
@@ -34,11 +35,16 @@
 
     private final String cpuCores;
     private final String coresPerSocket;
+    private final String cpuLimitMHz;
+    private final ResourceAllocationInfo cpuReservation;
```

Comment:
You've introduced data duplication here.
EITHER store the `cpuLimitMHz` only OR store the `cpuReservation` only ... but _do not_ store a `cpuReservation` that (always) contains a limit of `cpuLimitMHz`.

Personally, I'd recommend having only a field of `cpuLimitMHz` and pushing the code that creates a `ResourceAllocationInfo` down to `reconfigureCPU` where it's used.